### ISSUE TITLE
Install Failure on RHEL  'unicode' object has no attribute 'get'

### ISSUE THREAD

**Issue by uday31in** [microsoft/OMS-Agent-for-Linux#834]
OS Info:
```
  Operating System: Red Hat Enterprise Linux Server 7.4 (Maipo)
       CPE OS Name: cpe:/o:redhat:enterprise_linux:7.4:GA:server
            Kernel: Linux 3.10.0-693.11.6.el7.x86_64
      Architecture: x86-64
```

Log:
```
2018/10/27 19:08:15 [Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.7.9] Config decoded correctly.
2018/10/27 19:08:15 ERROR:Install failed with error: 'unicode' object has no attribute 'get'
2018/10/27 19:08:15 ERROR:Stacktrace: Traceback (most recent call last):
2018/10/27 19:08:15 ERROR:  File "/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.7.9/omsagent.py", line 157, in main
2018/10/27 19:08:15 ERROR:    exit_code = operations[operation]()
2018/10/27 19:08:15 ERROR:  File "/var/lib/waagent/Microsoft.EnterpriseCloud.Monitoring.OmsAgentForLinux-1.7.9/omsagent.py", line 242, in install
2018/10/27 19:08:15 ERROR:    workspaceId = public_settings.get('workspaceId')
2018/10/27 19:08:15 ERROR:AttributeError: 'unicode' object has no attribute 'get'
```
@Krnese

**Comment by shpimpal** [comment]
Thanks for reporting. I have created a task to investigate this in our backlog. Thanks!

**Comment by greg-pendlebury** [comment]
FWIW I had the same issue today because of the recent 'Azure Service Alert' about OMS crashing RHEL VMs. The instructions for performing the 'fix' requests we delete OMS 1.12.5 and reinstall 1.11.5 using the '--no-auto-upgrade', and gave the command as:
```
az vm extension set --name OMSAgentForLinux --publisher Microsoft.EnterpriseCloud.Monitoring --version 1.11.15 --protected-settings '{"workspaceKey":"myWorkspaceKey"}' --settings '{"workspaceId":"myWorkspaceId"}' --resource-group myResourceGroup --vm-name myVM --no-auto-upgrade
```

The important part is:
```
--protected-settings '{"workspaceKey":"myWorkspaceKey"}' --settings '{"workspaceId":"myWorkspaceId"}'
```
because this fails on Windows command line and results in a opaque error message that tells you nothing. You eventually trace it back to `/var/log/waagent.log` and find the error report by the OP:
```
... workspaceId = public_settings.get('workspaceId')
...  AttributeError: 'unicode' object has no attribute 'get'
```

I would say that the Python code needs some more defensive coding to prevent 'bad' config from entering the system and then report a meaningful error back through the install pipeline. The current message (`Non-zero exit code: 1`) is pretty bad.

The correct config for this particular use case/OS/CLI would be:
```
--protected-settings "{\"workspaceKey\":\"myWorkspaceKey\"}" --settings "{\"workspaceId\":\"myWorkspaceId\"}"
```
